frame/frame-layer.go

```diff
@@ -60,6 +60,19 @@ func NewFrameLayer(transportLayer io.ReadWriter) *Layer {
 	go frameLayer.bgWork()
 	go frameLayer.bgRead()
 
+	// Empty the buffer
+empty:
+	for {
+		select {
+		case <-parserInput:
```

Comment:
the issue is to make something that is non-blocking and that was the best solution I could think of.. because when the buffer is empty the read will block until new data is sent from the controller.